File: go-client-sdk/pkg/models/operations/getonecrmdeal.go
package operations

import (
	"github.com/speakeasy-sdks/orb-sdks/go-client-sdk/pkg/models/shared"
)

type GetOneCrmDealQueryParams struct {
	AccessToken string `queryParam:"style=form,explode=true,name=accessToken"`
	AllFields   *bool  `queryParam:"style=form,explode=true,name=allFields"`
	ID          string `queryParam:"style=form,explode=true,name=id"`
}

type GetOneCrmDealResponseBody struct {
	Deal *shared.Deal `json:"deal,omitempty"`
}

type GetOneCrmDealRequest struct {
	QueryParams GetOneCrmDealQueryParams
}

type GetOneCrmDealResponse struct {
	ContentType  string
	StatusCode   int64
	ResponseBody *GetOneCrmDealResponseBody
}

File: typescript-client-sdk/src/sdk/models/operations/getonecrmdeal.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class GetOneCrmDealQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=accessToken" })
  accessToken: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=allFields" })
  allFields?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=id" })
  id: string;
}


export class GetOneCrmDealResponseBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=deal" })
  deal?: shared.Deal;
}


export class GetOneCrmDealRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: GetOneCrmDealQueryParams;
}


export class GetOneCrmDealResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  responseBody?: GetOneCrmDealResponseBody;
}
